src/lib.rs

```diff
@@ -138,6 +139,77 @@ impl BorderStyle {
     }
 }
 
+pub struct PrinterBuilder<'a, Writer: Write> {
+    writer: &'a mut Writer,
+    show_color: bool,
+    show_char_panel: bool,
+    show_position_panel: bool,
+    border_style: BorderStyle,
+    use_squeeze: bool,
+    panels: u16,
+}
+
+impl<'a, Writer: Write> PrinterBuilder<'a, Writer> {
+    pub fn new(writer: &'a mut Writer) -> Self {
+        PrinterBuilder {
+            writer,
+            show_color: true,
+            show_char_panel: true,
+            show_position_panel: true,
+            border_style: BorderStyle::Unicode,
+            use_squeeze: true,
+            panels: 2,
+        }
+    }
+
+    pub fn with_writer(mut self, writer: &'a mut Writer) -> PrinterBuilder<'a, Writer> {
+        self.writer = writer;
+        self
+    }
```

Comment:
Why is this needed?

Reply:
Leftover from a previous pattern where the writer was optional. Thanks for noticing it!